### PR TITLE
ts/animations

### PR DESCRIPTION
**Pull request by marvin19**
Created animation folder and extracted animation functions and Fx class into their own files. Added type definition files.

### CODE REVIEW

**Review by TorsteinHonsi — APPROVED** [review]
Thanks!

**Comment by TorsteinHonsi** [comment]
Pending @bre1470 's approval

File: ts/Core/Animation/FxLike.d.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+ * */
+
+/* *
+ *
+ *  Declarations
+ *
+ * */
+export interface FxLike {
+
+}
```
Comment (bre1470) [inline ts/Core/Animation/FxLike.d.ts:18]:
```ts
/**
 * Optional properties for the Fx class. Should be defined outside of this file by `declare module` pattern.
 */
export interface FxLike {
    // nothing here yet
}
```

**Review by bre1470 — CHANGES_REQUESTED** [review]
Looks good. 👍

Though `FxLike` needs some explanation for others to understand the purpose.

**Review by bre1470 — APPROVED** [review]

**Pull request merged**